### PR TITLE
Add `since` parameter for public repositories

### LINKED ISSUE #686: List all public repositories
Reference: https://developer.github.com/v3/repos/#list-all-public-repositories 

- [x] `GetAllPublic()`
- [x]  `GetAllPublic(int since)`

### PR DESCRIPTION
**Pull request by alfhenrik**
Fixes #686 part deux

### CODE REVIEW

**Comment by alfhenrik** [comment]
This is ready for review

cc @shiftkey

File: Octokit.Tests.Integration/Reactive/ObservableRepositoriesClientTests.cs [hunk 1]
```diff
@@ -27,5 +28,26 @@ public class TheGetMethod
                 Assert.False(repository2.Fork);
             }
         }
+
+        public class TheGetAllPublicSinceMethod
+        {
+            [IntegrationTest]
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Reactive/ObservableRepositoriesClientTests.cs]: I think this test is going to run forever, but I'll mute this as part of preparing the next release

File: Octokit.Tests.Integration/Reactive/ObservableRepositoriesClientTests.cs [hunk 2]
```diff
@@ -27,5 +28,26 @@ public class TheGetMethod
...
+                var github = Helper.GetAuthenticatedClient();
+
+                var client = new ObservableRepositoriesClient(github);
+                var request = new PublicRepositoryRequest
+                {
+                    Since = 32732250
+                };
+                var repositories = await client.GetAllPublic(request).ToArray();
+                Assert.NotNull(repositories);
+                Assert.True(repositories.Any());
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Reactive/ObservableRepositoriesClientTests.cs]: :lipstick: `Assert.NotEmpty` here to make these two statements redundant

**Comment by shiftkey** [comment]
Looks good, just a couple of minor things.

File: Octokit.Tests.Integration/Reactive/ObservableRepositoriesClientTests.cs (hunk 1 above)
Comment (alfhenrik) [inline Octokit.Tests.Integration/Reactive/ObservableRepositoriesClientTests.cs]: Currently, yes it most likely will :smile: but when I created the test, it ran for a few seconds, so I can at least say I tested the feature :grimacing:

**Comment by alfhenrik** [comment]
Test updated and muted

File: Octokit.Tests/Reactive/ObservableRepositoriesClientTests.cs [hunk 3]
```diff
@@ -158,6 +158,61 @@ public class TheGetAllForCurrentMethod
...
+                    });
+                var gitHubClient = Substitute.For<IGitHubClient>();
+                gitHubClient.Connection.Get<List<Repository>>(firstPageUrl,
+                    Arg.Is<Dictionary<string, string>>(d => d.Count == 1
+                        && d["since"] == "364"), null)
+                    .Returns(Task.Factory.StartNew<IApiResponse<List<Repository>>>(() => firstPageResponse));
+                gitHubClient.Connection.Get<List<Repository>>(secondPageUrl, null, null)
+                    .Returns(Task.Factory.StartNew<IApiResponse<List<Repository>>>(() => secondPageResponse));
+                gitHubClient.Connection.Get<List<Repository>>(thirdPageUrl, null, null)
+                    .Returns(Task.Factory.StartNew<IApiResponse<List<Repository>>>(() => lastPageResponse));
```
Comment (shiftkey) [inline Octokit.Tests/Reactive/ObservableRepositoriesClientTests.cs]: :lipstick: `Task.FromResult` here there and everywhere?

File: Octokit/Models/Request/PublicRepositoryRequest.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class PublicRepositoryRequest : RequestParameters
+    {
+        public PublicRepositoryRequest()
+        {
+        }
+
+        public long Since { get; set; }
```
Comment (shiftkey) [inline Octokit/Models/Request/PublicRepositoryRequest.cs]: Would it be worth making this a required parameter?

**Comment by shiftkey** [comment]
A couple of questions :dog:

File: Octokit/Models/Request/PublicRepositoryRequest.cs (hunk 4 above)
Comment (alfhenrik) [inline Octokit/Models/Request/PublicRepositoryRequest.cs]: yep, certainly as if you wanted to find **ALL** public repos you'd just call the parameterless `GetAllPublic` method :see_no_evil:

**Comment by alfhenrik** [comment]
Questions answered :fist:

**Comment by shiftkey** [comment]
Thanks!

**Pull request merged**